helpers/formValidation.tsx

```diff
@@ -0,0 +1,35 @@
+import * as Yup from 'yup'
+
+const TEXT_MIN = 2
+const TEXT_MAX = 20
+const PASSWORD_MIN = 6
+const REGEX_ALPHANUMERICS = /^[a-zA-Z0-9_]*$/
+
+const signupValidation = Yup.object({
+  email: Yup.string()
+    .email('Invalid email address')
+    .required('Required'),
+  userName: Yup.string()
+    .strict(true)
+    .lowercase('Must be a lowercase string')
+    .matches(REGEX_ALPHANUMERICS, 'Must be alphanumerics characters')
+    .min(TEXT_MIN, `Must be at least ${TEXT_MIN} characters`)
+    .max(TEXT_MAX, `Must be ${TEXT_MAX} characters or less`)
+    .required('Required'),
+  password: Yup.string()
+    .min(PASSWORD_MIN, `Must be at least ${PASSWORD_MIN} characters`)
+    .max(TEXT_MAX, `Must be ${TEXT_MAX} characters or less`)
+    .required('Required'),
+  firstName: Yup.string()
+    .matches(REGEX_ALPHANUMERICS, 'Must be alphanumerics characters')
```

Comment:
my wife's first name has a space.

Reply:
@song Hum because we can't ask her name to change her name, I updated the regex for the sake of your wife. 😁
I could not find a way to accept only one space between words. Is anyone would have an idea?